src/ledger-db/initdb/1_create_transactions.sh

```diff
@@ -14,53 +14,98 @@
 # limitations under the License.
 
 
-# create default transactions in the ledger for the default user account.
+# create demo transactions in the ledger for the demo user accounts.
 # to make the history look realistic, we use biweekly large deposits,
 # followed by periodic small payments to random accounts.
 # values are chosen so that the depsoit in a period > payments in the same period
 set -u
 
 
 # skip adding transactions if not enabled
-if [ "$USE_DEFAULT_DATA" != "True"  ]; then
-    echo "no default transactions added"
+if [ "$USE_DEMO_DATA" != "True"  ]; then
+    echo "no demo transactions added"
     exit 0
 fi
 
 
+# Expected environment variables
+readonly ENV_VARS=(
+  "POSTGRES_DB"
+  "POSTGRES_USER"
+  "LOCAL_ROUTING_NUM"
+)
+
+
 add_transaction() {
     DATE=$(date -u +"%Y-%m-%d %H:%M:%S.%3N%z" --date="@$(($6))")
-    echo "adding default transaction: $1 -> $2"
+    echo "adding demo transaction: $1 -> $2"
     psql -X -v ON_ERROR_STOP=1 -v fromacct="$1" -v toacct="$2" -v fromroute="$3" -v toroute="$4" -v amount="$5" --username "$POSTGRES_USER" --dbname "$POSTGRES_DB" <<-EOSQL
         INSERT INTO TRANSACTIONS (FROM_ACCT, TO_ACCT, FROM_ROUTE, TO_ROUTE, AMOUNT, TIMESTAMP)
         VALUES (:'fromacct', :'toacct', :'fromroute', :'toroute', :'amount', '$DATE');
 EOSQL
 }
 
 
-PAY_PERIODS=3
-DAYS_BETWEEN_PAY=14
-SECONDS_IN_PAY_PERIOD=$(( 86400 * $DAYS_BETWEEN_PAY  ))
-echo $SECONDS_IN_PAY_PERIOD
-DEPOSIT_AMOUNT=250000
-PAYMENT_ACCOUNTS=("$DEFAULT_CONTACT_ACCOUNT_A" "$DEFAULT_CONTACT_ACCOUNT_B")
+create_transactions() {
+    PAY_PERIODS=3
+    DAYS_BETWEEN_PAY=14
+    SECONDS_IN_PAY_PERIOD=$(( 86400 * $DAYS_BETWEEN_PAY  ))
+    DEPOSIT_AMOUNT=250000
+
+    # create a UNIX timestamp in seconds since the Epoch
+    START_TIMESTAMP=$(( $(date +%s) - $(( $(($PAY_PERIODS+1)) * $SECONDS_IN_PAY_PERIOD  ))  ))
+
+    for i in $(seq 1 $PAY_PERIODS); do
+        # create deposit transaction for each user
+        for account in ${USER_ACCOUNTS[@]}; do
+            add_transaction "$EXTERNAL_ACCOUNT" "$account" "$EXTERNAL_ROUTING" "$LOCAL_ROUTING_NUM" $DEPOSIT_AMOUNT $START_TIMESTAMP
+        done
 
-# create a UNIX timestamp in seconds since the Epoch
-START_TIMESTAMP=$(( $(date +%s) - $(( $(($PAY_PERIODS+1)) * $SECONDS_IN_PAY_PERIOD  ))  ))
+        # create payments between users
+        TRANSACTIONS_PER_PERIOD=$(shuf -i 10-20 -n1)
```

Comment:
If I'm understanding this right, if you get extremely unlucky and select the same `SENDER_ACCOUNT` for 20 payments each of $250, you'll end up spending $5000. This is more than the $2500 deposited in a pay period, so you could potentially create a negative balance.

Maybe you should adjust the numbers or logic so that this isn't possible?

Reply:
Good point.  Adjusted so that an account will never be overdrawn.